app/controllers/AdditionalMetadataController.scala

```diff
@@ -25,4 +45,106 @@ class AdditionalMetadataController @Inject () (val consignmentService: Consignme
+          consignmentId,
+          request.token.name,
+          parentFolder,
+          totalPages,
+          limit,
+          page,
+          selectedFolderId,
+          navigationForm.fill(NodesFormData(nodesToDisplay, previouslySelectedIds, selected = List(), allNodes = List(), page, selectedFolderId.toString)))
+        )
+      }
+  }
+
+  def submit(consignmentId: UUID, page: Int, limit: Option[Int], selectedFolderId: UUID): Action[AnyContent] = standardTypeAction(consignmentId)
+  { implicit request: Request[AnyContent] =>
+    val errorFunction: Form[NodesFormData] => Future[Result]  = { formWithErrors: Form[NodesFormData] =>
+      consignmentService.getConsignmentPaginatedFile(consignmentId, page - 1, limit, selectedFolderId, request.token.bearerAccessToken)
+        .map { paginatedFiles =>
+          Ok(views.html.standard.additionalMetadata(
+            consignmentId,
+            request.token.name,
+            paginatedFiles.parentFolder.get,
+            paginatedFiles.paginatedFiles.totalPages.get,
+            limit,
+            page,
+            selectedFolderId,
+            formWithErrors)
+          )
+        }
+    }
+
+    val successFunction: NodesFormData => Future[Result] = { formData: NodesFormData =>
+      val pageToGo = formData.pageSelected
+      val folderSelected = formData.folderSelected
+      //PageToGo -1 so that the pagination number matches that of the page in the url
+      consignmentService.getConsignmentPaginatedFile(consignmentId, pageToGo -1, limit, UUID.fromString(folderSelected), request.token.bearerAccessToken)
+        .map { paginatedFiles =>
+          val edges: List[PaginatedFiles.Edges] = paginatedFiles.paginatedFiles.edges.get.flatten
+          val selected = formData.selected
+          val previouslySelectedIds: String = formData.previouslySelected + "," + selected.mkString(",")
+          val updatedSelection = handleDeselection(selected, formData.allNodes, previouslySelectedIds)
+          val nodesToDisplay = generateNodesToDisplay(edges, updatedSelection)
+
+          Ok(views.html.standard.additionalMetadata(
+            consignmentId,
+            request.token.name,
+            paginatedFiles.parentFolder.get,
+            paginatedFiles.paginatedFiles.totalPages.get,
+            limit,
+            pageToGo,
+            selectedFolderId,
+            navigationForm.fill(NodesFormData(nodesToDisplay, updatedSelection.mkString(","), List(), List(), pageToGo, folderSelected)))
+          )
+        }
+    }
+
+    val formValidationResult: Form[NodesFormData] = navigationForm.bindFromRequest()
+    formValidationResult.fold(
+      errorFunction,
+      successFunction
+    )
+  }
+
+  private def handleDeselection(selected: List[String], allNodes: List[String], previouslySelectedIds: String): Set[String] = {
+    //Filter out selected Nodes
+    val allNodesWithoutSelection = allNodes.filter(x => !selected.contains(x)).mkString(",")
```

Comment:
Just a style thing, but I wouldn't use `x` as a val. Would prefer to use `s` here given its a string.